### PR TITLE
Improved error handling for ModelVariables

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**
**List of other things done along the way:**

1. Added has_ATTR functionality for optional attributes without default values. This includes parsing + API functions + tests: Binary: maxSize, Clock: priority, intervalDecimal, resolution, intervalCounter
2. Added API functions to retrieve "reinit" attribute from float variables

**Error checking changes (currently only in ModelVariables, also affects parts of TypeDefinitions):**

1. Failure to parse a variable will no longer abort parsing of remaining variables.
2. Failure to parse an attribute will no longer suspend parsing of remaining attributes. Exceptions: Failure in common attributes
3. Failure to parse a child element will no longer suspend parsing other child element on the same level, e.g., Start, Dimension.
4. Error checks will no longer abort upon first failure, affects both error checks for invalid attribute combination and unique name/VR checks.
5. Failure to parse a variable or failed essential error checks causes parsing to exit with closing tag of ModelVariables. Currently, essential is almost anything marked as "error" (few exceptions)

**Changed error categories for almost all errors checks:**
1. Fatal: Malloc failures & exit at end of ModelVariables.
2. Error: Element fails to parse due to missing/invalid required attribute
3. Warning: Everything else, e.g. invalid attribute combinations, failure to parse optional attributes.

Rationale is that an error points to structural issues in the FMU generation tool, but we aim to parse as much as possible and return an object one can interact with, even if there are issues reported by warnings.

Elements failing to parse is primarily determined by the question of breaking the API, slightly inconsistent at times, e.g., Dimension without VR or value yields error, but passes, since API will still be safe, rationale here is the child element won't cause the parent to fail.

### CODE REVIEW

**Review by modelonrobinandersson — COMMENTED** [review]
Well done! This was a massive effort and I am glad that it is done, I only had a few minor nitpick comments and in particular a question about bitness on a few of the functions (unrelated to your changes).

File: src/Import/src/FMI3/fmi3_import_variable.c [hunk 1]
```diff
@@ -98,7 +99,7 @@ fmi3_import_variable_list_t* fmi3_import_get_variable_clocks(fmi3_import_t* fmu,
 // -----------------------------------------------------------------------------
 
 fmi3_import_float64_variable_t* fmi3_import_get_variable_as_float64(fmi3_import_variable_t* v) {
-    return fmi3_xml_get_variable_as_float64(v);
+    return v ? fmi3_xml_get_variable_as_float64(v) : NULL;
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply src/Import/src/FMI3/fmi3_import_variable.c:102]: I already captured it in our ticket system, along with adding more tests for these type of safety enhancements.

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 2]
```diff
@@ -1414,36 +1461,47 @@ int fmi3_xml_handle_Variable(fmi3_xml_parser_context_t* context, const char* dat
...
 
         /* Get vr, name and description */
-        if (fmi3_xml_parse_attr_as_uint32(context, elm_id, fmi_attr_id_valueReference, 1, &vr, 0))  return -1;
-        if (fmi3_xml_parse_attr_as_string(context, elm_id, fmi_attr_id_name,           1, bufName)) return -1;
-        if (fmi3_xml_parse_attr_as_string(context, elm_id, fmi_attr_id_description,    0, bufDesc)) return -1;
+
+        /* Get required attributes: VR, name */
+        int req = 0;
+        req |= fmi3_xml_parse_attr_as_uint32(context, elm_id, fmi_attr_id_valueReference, 1, &vr, 0);
+        req |= fmi3_xml_parse_attr_as_string(context, elm_id, fmi_attr_id_name,           1, bufName);
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1480]: Good point, added.

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 3]
```diff
@@ -1467,121 +1525,109 @@ int fmi3_xml_handle_Variable(fmi3_xml_parser_context_t* context, const char* dat
...
-            if (fmi3_xml_parse_attr_as_uint64(context, fmi3_xml_elmID_Dimension, attrId, 0, &dim->start, 0)) {
-                return -1;
-            }
-        } else if (hasVr) {
-            dim->has_vr = 1;
-            attrId = fmi_attr_id_valueReference;
-            if (fmi3_xml_parse_attr_as_uint32(context, fmi3_xml_elmID_Dimension, attrId, 0, &dim->vr, 0)) {
-                return -1;
-            }
+            ret = fmi3_xml_parse_attr_as_uint64(context, fmi3_xml_elmID_Dimension, fmi_attr_id_start, 0, &dim->start, 0);
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1624]: Added.

**Review by modelonrobinandersson — APPROVED** [review]

**Pull request merged**